pkg/deploy/kubernetes_image_puller.go

```diff
@@ -379,6 +385,88 @@ func UpdateImagePullerSpecIfEmpty(ctx *DeployContext) (orgv1.CheClusterSpecImage
 	return ctx.CheCluster.Spec.ImagePuller, nil
 }
 
+func AddRelatedImages(ctx *DeployContext) {
+	relatedImages := GetRelatedImagesEnvVars()
+
+	if len(relatedImages) == 0 {
+		return
+	}
+
+	if len(ctx.CheCluster.Spec.ImagePuller.Spec.Images) > 0 &&
+		!strings.HasSuffix(ctx.CheCluster.Spec.ImagePuller.Spec.Images, ";") {
+		ctx.CheCluster.Spec.ImagePuller.Spec.Images += ";"
+	}
+
+	for _, envVar := range relatedImages {
+		ctx.CheCluster.Spec.ImagePuller.Spec.Images += envVar.Name + "=" + envVar.Value + ";"
+	}
+	ctx.ClusterAPI.Client.Update(context.TODO(), ctx.CheCluster, &client.UpdateOptions{})
+}
+
+func GetRelatedImagesEnvVars() []v1.EnvVar {
+	relatedImagePatterns := [...]string{
+		"^RELATED_IMAGE_.*_plugin_java8$",
+		"^RELATED_IMAGE_.*_plugin_java11$",
+		"^RELATED_IMAGE_.*_plugin_kubernetes$",
+		"^RELATED_IMAGE_.*_plugin_openshift$",
+		"^RELATED_IMAGE_.*_plugin_broker.*",
+		"^RELATED_IMAGE_.*_theia.*",
+		"^RELATED_IMAGE_.*_stacks_cpp$",
+		"^RELATED_IMAGE_.*_stacks_dotnet$",
+		"^RELATED_IMAGE_.*_stacks_golang$",
+		"^RELATED_IMAGE_.*_stacks_php$",
+		"^RELATED_IMAGE_.*_cpp_.*_devfile_registry_image.*",
+		"^RELATED_IMAGE_.*_dotnet_.*_devfile_registry_image.*",
+		"^RELATED_IMAGE_.*_golang_.*_devfile_registry_image.*",
+		"^RELATED_IMAGE_.*_php_.*_devfile_registry_image.*",
+		"^RELATED_IMAGE_.*_java.*_maven_devfile_registry_image.*",
+	}
```

Comment:
@tolusha what do you think about this approach?

Reply:
Sounds good.